mantidproject/mantid
# Delete Engineering Diffraction interface wrapper on close

I introduced a bug in https://github.com/mantidproject/mantid/pull/31809 wherein you would get an error if you opened up the interface, closed it, and then tried to reopen in the same session. This was because the C++ object was being deleted but the wrapper was hanging around due to the global reference in the Engineering_Diffraction.py init script.

This ensures that global reference is cleared when the interface is closed.

**To test:**
1. Open the interface, close it, reopen it
2. It should reopen fine

*There is no associated issue.*

*This does not require release notes* because **it fixes a bug introduced this sprint**



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
